### PR TITLE
Closes #9757 remove downloads notification when private tabs are closed

### LINKED ISSUE #9757: Remove downloads notification when private tabs are closed
related issue https://github.com/mozilla-mobile/fenix/issues/11190

### PR DESCRIPTION
**Pull request by Amejia481**
We want to observe any actions that remove tabs, to verify if there are private downloads associated with it if so we want to remove their notifications from the status bar.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt [hunk 1]
```diff
@@ -160,6 +161,100 @@ class AbstractFetchDownloadServiceTest {
...
+        service.downloadJobs[downloadState.id] = downloadJobState
+
+        service.handleRemovePrivateDownloadIntent(downloadState)
+
+        verify(service).removeDownloadJob(downloadJobState)
+        verify(browserStore).dispatch(DownloadAction.RemoveDownloadAction(downloadState.id))
+    }
+
+    @Test
+    fun `WHEN handleRemovePrivateDownloadIntent is called with a none private download THEN removeDownloadJob must not be called`() {
```
Comment (csadilek) [inline components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt:239]: nit: with a non-private (or regular) download.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 2]
```diff
@@ -256,14 +256,35 @@ abstract class AbstractFetchDownloadService : Service() {
             store.state.downloads[it]
         } ?: return START_REDELIVER_INTENT
 
+        if (intent.action ?: "" == ACTION_REMOVE_PRIVATE_DOWNLOAD) {
+            handleRemovePrivateDownloadIntent(download)
```
Comment (csadilek) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:260]: We should also stop the download entirely here.

**Review by csadilek — COMMENTED** [review]
Looks good! A comment as discussed and some nits below.

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 2 above)
Comment (csadilek) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:260]: 🤔 We decided to handle that separately as we likely need to prompt the user for this.

**Review by csadilek — APPROVED** [review]
Just some nits below. 🚀

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 3]
```diff
@@ -256,14 +256,35 @@ abstract class AbstractFetchDownloadService : Service() {
             store.state.downloads[it]
         } ?: return START_REDELIVER_INTENT
 
+        if (intent.action ?: "" == ACTION_REMOVE_PRIVATE_DOWNLOAD) {
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:259]: 👍🏽

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 2 above)
Comment (Amejia481) [reply components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:260]: I filed https://github.com/mozilla-mobile/fenix/issues/18160 for this case

**Pull request merged**